kcmdline.c

```diff
@@ -48,12 +48,15 @@ EFI_STATUS check_cmdline(CONST CHAR8 *cmdline, UINTN cmdline_len) {
 	CHAR8 c = '\0';
 	CHAR8 *buf = NULL;
 	CHAR8 *tokens[MAX_TOKENS];
+	EFI_STATUS status = EFI_SUCCESS;
 	int i;
 	int start = -1;
 	int num_toks = 0;
 	buf = AllocatePool(cmdline_len);
-	if (!buf)
-		return EFI_OUT_OF_RESOURCES;
+	if (!buf) {
+		status = EFI_OUT_OF_RESOURCES;
```

Comment:
Wait -is it safe to FreePool(null)?  Maybe in this case you should just return?

Reply:
Good catch. thank you for reading. Fixed now.